File: sum_vectors.go
package main

import "fmt"

func main() {
	a := make([]int, 10_000_000, 10_000_000)
	b := make([]int, 10_000_000, 10_000_000)

	for i := 1; i <= 10_000_000; i++ {
		a[i-1] = i
		b[i-1] = i
	}

	for i := 0; i < len(a); i++ {
		a[i] += b[i]
	}

	fmt.Printf("last element = %d\n", a[len(a)-1])
}

File: sum_vectors.rs
fn main() {
    let mut a = (1..=10_000_000).map(|n| n as i32).collect::<Vec<_>>();
    let b = (1..=10_000_000).map(|n| n as i32).collect::<Vec<_>>();

    for i in 0..10_000_000 {
        a[i] += b[i];
    }

    println!("last element = {:?}", a[a.len()-1]);
}
